src/tests/simple_pipeline.c

```diff
@@ -0,0 +1,696 @@
+		iods[i].iod_nr		= 1;
+		iods[i].iod_size	= STRING_LEN;
+		iods[i].iod_recxs	= NULL;
+		iods[i].iod_type	= DAOS_IOD_SINGLE;
+		d_iov_set(&iods[i].iod_name, (void *) fields[i],
+						strlen(fields[i]));
+	}
+
+	/**
+	 * reading in chunks of 64 keys (at most) at a time
+	 */
+	nr_kds   = 64;
+	nr_iods  = NR_IODS;
+
+	/** sgl_keys: to store the retrieved dkeys */
+	kds		= malloc(sizeof(daos_key_desc_t)*nr_kds);
+	sgl_keys	= malloc(sizeof(d_sg_list_t)*nr_kds);
+	iovs_keys	= malloc(sizeof(d_iov_t)*nr_kds);
+	buf_keys	= malloc(nr_kds*STRING_LEN);
+	for (i = 0; i < nr_kds; i++) {
+		sgl_keys[i].sg_nr	= 1;
+		sgl_keys[i].sg_nr_out	= 0;
+		sgl_keys[i].sg_iovs	= &iovs_keys[i];
+		d_iov_set(&iovs_keys[i], &(buf_keys[i*STRING_LEN]), STRING_LEN);
+	}	
+
+	/** sgl_recx: to store the retrieved data for the akeys of each dkey */
+	sgl_recx	= malloc(sizeof(d_sg_list_t)*nr_kds*nr_iods);
+	iovs_recx	= malloc(sizeof(d_iov_t)*nr_kds*nr_iods);
+	buf_recx	= malloc(nr_kds*nr_iods*STRING_LEN);
+	for (i = 0; i < nr_kds; i++) {
+		for (j = 0; j < nr_iods; j++) {
+			l = i*nr_iods+j;
+			sgl_recx[l].sg_nr	= 1;
+			sgl_recx[l].sg_nr_out	= 0;
+			sgl_recx[l].sg_iovs	= &iovs_recx[l];
+			d_iov_set(&iovs_recx[l], &(buf_recx[l*STRING_LEN]),
+							STRING_LEN);
+		}
+	}
+	sgl_aggr	= malloc(sizeof(d_sg_list_t)*nr_aggr);
+	iovs_aggr	= malloc(sizeof(d_iov_t)*nr_aggr);
+	buf_aggr	= malloc(sizeof(double)*nr_aggr);
+	for (i = 0; i < nr_aggr; i++) {
+		sgl_aggr[i].sg_nr	= 1;
+		sgl_aggr[i].sg_nr_out	= 0;
+		sgl_aggr[i].sg_iovs	= &iovs_aggr[i];
+		d_iov_set(&iovs_aggr[i], (void *) &(buf_aggr[i]), sizeof(double));
+	}
+
+	/** reset anchor */
+	memset(&anchor, 0, sizeof(daos_anchor_t));
+
+	/** reading 64 records at a time */
+	while (!daos_anchor_is_eof(&anchor)) {
+		nr_kds = 64; /** trying to read 64 at a time */
+
+		rc = daos_pipeline_run(coh, oh, *pipeline, DAOS_TX_NONE, 0,
+				       NULL, &nr_iods, iods, &anchor, &nr_kds,
+				       kds, sgl_keys, sgl_recx, sgl_aggr,
+				       NULL);
+
+		ASSERT(rc == 0, "Pipeline run failed with %d", rc);
+		/** process nr_kds fetched records */
+		for (i = 0; i < nr_kds; i++) {
+			char *dkey = (char *) sgl_keys[i].sg_iovs->iov_buf;
+			printf("\tname(dkey)=%s%*c", dkey,
+					(int)(STRING_LEN-strlen(dkey)), ' ');
+			for (j = 0; j < nr_iods-1; j++) {
+				char *akey;
+				l = i*nr_iods+j;
+				akey = (char *) sgl_recx[l].sg_iovs->iov_buf;
+				printf("%.*s(akey)=%s%*c",
+					 (int)    iods[j].iod_name.iov_buf_len,
+					 (char *) iods[j].iod_name.iov_buf,
+					 akey,
+					 (int)(STRING_LEN-strlen(akey)), ' ');
+			}
+			int *akey;
+			l = i*nr_iods+(nr_iods-1);
+			akey = (int *) sgl_recx[l].sg_iovs->iov_buf;
+			printf("%.*s(akey)=%d\n",
+				 (int)    iods[nr_iods-1].iod_name.iov_buf_len,
+				 (char *) iods[nr_iods-1].iod_name.iov_buf,
+				 *akey);
+		}
+	}
+	printf("\n");
+	for (i = 0; i < nr_aggr; i++) {
+		double *res = (double *) sgl_aggr[i].sg_iovs->iov_buf;
+		printf("  ---result[%u]=%f---\n",i,*res);
```

Comment:
(style)  Missing a blank line after declarations
(style)  space required after that ',' (ctx:VxV)
(style)  space required after that ',' (ctx:VxO)
(style)  space required before that '*' (ctx:OxV)